server/src/test/java/org/opensearch/search/approximate/ApproximatePointRangeQueryTests.java

```diff
@@ -29,54 +35,277 @@
+            }
+
+            @Override
+            SortField.Type getSortFieldType() {
+                return SortField.Type.DOUBLE;
+            }
+        },
+        UNSIGNED_LONG("unsigned_long_field", ApproximatePointRangeQuery.UNSIGNED_LONG_FORMAT) {
+            @Override
+            byte[] encode(Number value) {
+                byte[] bytes = new byte[BigIntegerPoint.BYTES];
+                BigIntegerPoint.encodeDimension(BigInteger.valueOf(value.longValue()), bytes, 0);
+                return bytes;
+            }
+
+            @Override
+            void addField(Document doc, String fieldName, Number value) {
+                doc.add(new BigIntegerPoint(fieldName, BigInteger.valueOf(value.longValue())));
+            }
+
+            @Override
+            void addDocValuesField(Document doc, String fieldName, Number value) {
+                doc.add(new NumericDocValuesField(fieldName + "_sort", value.longValue()));
+            }
+
+            @Override
+            Query rangeQuery(String fieldName, Number lower, Number upper) {
+                return BigIntegerPoint.newRangeQuery(
+                    fieldName,
+                    BigInteger.valueOf(lower.longValue()),
+                    BigInteger.valueOf(upper.longValue())
+                );
+            }
+
+            @Override
+            SortField.Type getSortFieldType() {
+                return SortField.Type.LONG;
+            }
+
+            @Override
+            String getSortFieldName() {
+                return fieldName + "_sort";
+            }
+        };
+
+        final String fieldName;
+        final Function<byte[], String> format;
+
+        NumericType(String fieldName, Function<byte[], String> format) {
+            this.fieldName = fieldName;
+            this.format = format;
+        }
+
+        abstract byte[] encode(Number value);
+
+        abstract void addField(Document doc, String fieldName, Number value);
+
+        abstract void addDocValuesField(Document doc, String fieldName, Number value);
+
+        abstract Query rangeQuery(String fieldName, Number lower, Number upper);
+
+        abstract SortField.Type getSortFieldType();
+
+        String getSortFieldName() {
+            return fieldName;
+        }
+
+        long getMinTestValue() {
+            return this == UNSIGNED_LONG ? 0 : -100;
+        }
+    }
 
     public void testApproximateRangeEqualsActualRange() throws IOException {
         try (Directory directory = newDirectory()) {
             try (RandomIndexWriter iw = new RandomIndexWriter(random(), directory, new WhitespaceAnalyzer())) {
                 int dims = 1;
-
-                long[] scratch = new long[dims];
-                for (int i = 0; i < 100; i++) {
-                    int numPoints = RandomNumbers.randomIntBetween(random(), 1, 10);
+                int numDocs = RandomNumbers.randomIntBetween(random(), 50, 200);
```

Comment:
`numDocs` should be larger than 512 to ensure that `bkd` has more than one leaf node? Otherwise, the optimization point may not be tested. The same problem also exists in other uts, such as `testIntersectRightSingleChildNode`

Reply:
Thats a good point, let me check this.